types/indexer_block_events.go

```diff
@@ -0,0 +1,93 @@
+package types
+
+import (
+	"time"
+)
+
+type IndexerBlockEventManager struct {
+	height   uint32
+	time     time.Time
+	txHashes []string
+	// map from tx hash to list of tx events
+	txEventsMap map[string][]*IndexerTendermintEvent
+}
+
+// NewIndexerBlockEventManager returns a new IndexerBlockEventManager.
+// This should be called in BeginBlocker.
+func NewIndexerBlockEventManager(height uint32, time time.Time) *IndexerBlockEventManager {
+	return &IndexerBlockEventManager{
+		txHashes:    []string{},
+		txEventsMap: make(map[string][]*IndexerTendermintEvent),
+		height:      height,
+		time:        time,
+	}
+}
+
+// MergeEvents merges the events from another block event manager into this one if
+// the block heights are the same.
+func (eventManager *IndexerBlockEventManager) MergeEvents(mgr *IndexerBlockEventManager) {
+	if eventManager.height != mgr.height {
```

Comment:
When would this be expected to happen, and why does the `EventManager` not have to check for this? Additionally, would this be an error case or would it be expected during normal operation?